writing-visualising/dashboards_rshiny.qmd

```diff
@@ -7,36 +7,148 @@ title: "Creating R Shiny dashboards"
+
+In the example below, we demonstrate two examples. `unlazy_data` is all being used so we load it into the environment to be shared between multiple users. `lazy_school_table` is being used by one school at a time so is probably more efficient to load in only one school than all schools.
+
+```{r eval = FALSE}
+# NOT RUN
+if(FALSE){
+  # create a pool connection
+  connection <- pool::dbPool(...)
+  
+  # unlazy data collected into memory BEFORE server
+  unlazy_data <- dplyr::tbl(connection, DBI::Id(...)) |>
+    collect()
+  
+  # create a lazy connection to a database
+  lazy_school_table <- dplyr::tbl(connection, DBI::Id(...))
+  
+  # tell shiny to close the connection when the app stops
+  shiny::onStop(function() poolClose(connection))
+  
+  # an example ui to select a school
+  ui <- shiny::fluidPage(
+    shiny::selectInput(
+      inputId = "school"
+    ),
+    shiny::tableOutput("all_schools"),
+    shiny::tableOutput("one_school")
+  )
+  
+  # data collected into memory ON SERVER SIDE
+  server <- function(input, output, session){
+    school_level_data <- reactive({
+      lazy_school_table |>
+        filter(school_name == !!input$school) |>
+        collect()
+    })
+    
+    # table showing all schools
+    output$all_schools <- renderTable({
+      unlazy_data
+    })
+    
+    # table showing filtered school
+    output$one_school <- renderTable({
+      school_level_data()
+    })
+  }
+}
+```
+
+### Levelling up your shiny
+
+This is just the tip of the iceberg. 
+
+[Mastering Shiny](https://mastering-shiny.org/index.html) is a really useful read, especially the [Best Practice](https://mastering-shiny.org/scaling-intro.html) section.
+
+There is a plethora of [awesome shiny extensions](https://github.com/nanxstats/awesome-shiny-extensions) to explore. Keep it accessible, use packages from CRAN/ones you trust and the world is your oyster.
+
+For more **custom user interface**: 
+
+- have a read of [Outstanding User Interfaces with Shiny](https://unleash-shiny.rinterface.com/welcome.html) for how the web stuff works with shiny 
+- for easy theme customisation [Bootstrap Live Customiser](https://www.bootstrap-live-customizer.com/) 
+- For more context of the web bits, [HTML](https://www.w3schools.com/html/default.asp), [CSS](https://www.w3schools.com/css/default.asp) and [Bootstrap](https://www.w3schools.com/bootstrap/bootstrap_ver.asp) give a better understanding of what's under the hood. 
+
+:::{.callout-caution}
+HTML, CSS and Bootstrap are a slippery slope into web development. Remember that if you're working with Shiny then good analysis comes first!
+:::
+
+For **interactive maps**, check out [this section](../learning-development/r.html#example-shiny-app-with-map).
```

Comment:
'this section' isn't great link text for accessibility, could you make it more like 'this section on interactive maps' ?